### ISSUE TITLE
KNN traditional decluttering

### ISSUE THREAD

**Issue by pepecoin1** [hanelo/KNN-Traditional-KNN-bayesian#1]
the [knn traditional ](https://github.com/hanelo/KNN-Traditional-KNN-bayesian/blob/839e25348e90b37d5c622c580d5e27a3fc4bdb73/KNN%20traditional%20(1).ipynb)file could be decluttered:

```
import numpy as np
from sklearn.datasets import load_iris
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score

def traditional_knn(x_train, y_train, x_test, k):
    knn = KNeighborsClassifier(n_neighbors=k)
    knn.fit(x_train, y_train)
    test_predictions = knn.predict(x_test)
    return test_predictions

def evaluate_classification(y_true, y_pred):
    accuracy = accuracy_score(y_true, y_pred)
    precision = precision_score(y_true, y_pred, average=None)
    recall = recall_score(y_true, y_pred, average=None)
    f1 = f1_score(y_true, y_pred, average=None)
    return accuracy, precision, recall, f1

# Load the Iris dataset
iris = load_iris()
X = iris.data
y = iris.target

# Set the value of k
k = 3

# Split the dataset into training and test sets
np.random.seed(42)  # Set seed for reproducibility
indices = np.random.permutation(len(X))
train_size = int(0.8 * len(X))

X_train, X_test = X[indices[:train_size]], X[indices[train_size:]]
y_train, y_test = y[indices[:train_size]], y[indices[train_size:]]

# Run traditional k-NN with MAP estimation
traditional_test_predictions = traditional_knn(X_train, y_train, X_test, k)

# Compute evaluation metrics for traditional k-NN
traditional_test_accuracy, traditional_test_precision, traditional_test_recall, traditional_test_f1_score = evaluate_classification(y_test, traditional_test_predictions)

print("Traditional k-NN Results:")
print(f"Test Accuracy: {traditional_test_accuracy}")
print(f"Test Precision: {traditional_test_precision}")
print(f"Test Recall: {traditional_test_recall}")
print(f"Test F1-score: {traditional_test_f1_score}")
```

- introducted evaluate_classification function to calculate accuracy, precision, recall, and F1-score. makes code more readable.
- set a seed (np.random.seed(42)) for reproducibility of random data shuffling.
- now uses numpy.random.permutation for more efficient data splitting into training and test sets.
